File: services/bff/src/application/dtos/agent.dto.ts
import { AgentEntity } from '../../domain/entities/agent.entity';

export interface AgentOutput {
  id: string;
  conversationId: string;
  name: string;
}

export const toAgentOutput = (agent: AgentEntity): AgentOutput => ({
  id: agent.id,
  conversationId: agent.conversationId,
  name: agent.name,
});

export interface GetAgentsByConversationIdsDto {
  conversationIds: string[];
}

File: services/agent/src/application/dtos/agent_dto.py
from dataclasses import dataclass

from src.domain.entities.agent import Agent


@dataclass
class AgentOutput:
    id: str
    conversation_id: str
    name: str
    system_prompt: str
    model: str
    created_at: str
    updated_at: str

    @staticmethod
    def from_entity(agent: Agent) -> "AgentOutput":
        return AgentOutput(
            id=agent.id,
            conversation_id=agent.conversation_id,
            name=agent.name,
            system_prompt=agent.system_prompt,
            model=agent.model,
            created_at=agent.created_at.isoformat(),
            updated_at=agent.updated_at.isoformat(),
        )
